core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -434,53 +435,57 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
+            // actually sending an active ack for this action. This prevents an active ack response for the activation
+            // being dropped because the listener is not yet active (this can happen if the response is faster coming back
+            // to the controller than the future callbacks getting a chance to run) and forcing the db polling fall back to kick in.
+            val (aid, callback) = waitForActivationResponse(user.namespace, ActivationId(), timeout)
+            (aid, Some(callback))
+        } else (ActivationId(), None)
+
+        val message = Message(transid, s"/actions/invoke/${action.namespace}/${action.name}/${action.rev}", user, activationId, args)
         info(this, s"[POST] action activation id: ${message.activationId}")
-        performLoadBalancerRequest(INVOKER, message, transid) map {
-            (action.limits.timeout(), _)
-        } flatMap {
-            case (duration, response) =>
-                response.id match {
-                    case Some(activationId) =>
-                        Future successful (duration, activationId)
-                    case None =>
-                        if (response.error.getOrElse("??").equals("too many concurrent activations")) {
-                            // DoS throttle
-                            warn(this, s"[POST] action activation rejected: ${response.error.getOrElse("??")}")
-                            Future failed new TooManyActivationException("too many concurrent activations")
-                        } else {
-                            error(this, s"[POST] action activation failed: ${response.error.getOrElse("??")}")
-                            Future failed new IllegalStateException(s"activation failed with error: ${response.error.getOrElse("??")}")
-                        }
+
+        performLoadBalancerRequest(INVOKER, message, transid) flatMap { response =>
+            response.id match {
+                case Some(activationId) =>
+                    Future successful activationId
+                case None =>
+                    if (response.error.getOrElse("??").equals("too many concurrent activations")) {
+                        // DoS throttle
+                        warn(this, s"[POST] action activation rejected: ${response.error.getOrElse("??")}")
+                        Future failed new TooManyActivationException("too many concurrent activations")
+                    } else {
+                        error(this, s"[POST] action activation failed: ${response.error.getOrElse("??")}")
+                        Future failed new IllegalStateException(s"activation failed with error: ${response.error.getOrElse("??")}")
+                    }
+            }
+        } flatMap { activationId =>
+            blockingCallback match {
+                case Some(callback) => {
+                    // Blocking invocation
+                    callback map { activation =>
```

Comment:
On a timeout the Future `callback` is failing.
On a timeout, in line 524 a `BlockingInvokeTimeout` is thrown. This will be passed through and catched in line 276.